src/Microsoft.Deployment.DotNet.Releases/src/ReleaseVersion.cs

```diff
@@ -693,144 +670,247 @@ internal static bool IsPrereleaseIdentifier(string input)
+                    throw new FormatException(string.Format(ReleasesResources.InvalidPrerelease, prerelease));
+                }
 
-                return true;
+                return false;
             }
-            catch
+
+            if (!IsValidBuildMetadata(build))
             {
-                if (canThrow)
+                if (throwOnFailure)
                 {
-                    throw;
+                    throw new FormatException(string.Format(ReleasesResources.InvalidBuildMetadata, build));
                 }
 
                 return false;
             }
+
+            version = new ReleaseVersion
+            {
+                Major = versions[0],
+                Minor = versions[1],
+                Patch = versions[2],
+                Prerelease = prerelease,
+                BuildMetadata = build,
+                SdkFeatureBand = (versions[2] / 100) * 100,
+                SdkPatchLevel = versions[2] % 100
+            };
+
+            return true;
         }
 
         /// <summary>
         /// Tries to convert a part of the core version to an integer.
         /// </summary>
         /// <param name="input">A string containing the major, minor, or patch value of the version.</param>
-        /// <param name="canThrow">A boolean indicating whether an exception should be raised if the conversion failed.</param>
+        /// <param name="throwOnFailure">A boolean indicating whether an exception should be raised if the conversion failed.</param>
         /// <param name="value">The converted value.</param>
         /// <returns><see langword="true"/> if the string could be converted; <see langword="false"> otherwise.</see> </returns>
-        /// <exception cref="FormatException" />
-        /// <exception cref="ArgumentOutOfRangeException" />
-        internal static bool TryParseCoreVersionPart(string input, bool canThrow, out int value)
+        /// <exception cref="FormatException">If the input string is an invalid major, minor or patch value.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">If <paramref name="throwOnFailure"/> is <see langword="true"/> and the value is less than 0.</exception>
+        /// <exception cref="OverflowException">If <paramref name="throwOnFailure"/> is <see langword="true"/> and the value exceeds <see cref="int.MaxValue"/>.</exception>
+        internal static bool TryParseCoreVersionPart(string input, bool throwOnFailure, out int value)
         {
             value = -1;
 
-            try
+            if (!IsNumericIdentifier(input))
             {
-                if (!IsNumericIdentifier(input))
+                if (throwOnFailure)
                 {
                     throw new FormatException(string.Format(ReleasesResources.InvalidNumericIdentifier, input));
                 }
 
+                return false;
+            }
+
+            if (throwOnFailure)
+            {
                 value = int.Parse(input);
+            }
+            else
```

Comment:
Consider utilizing an `else if` to reduce the level of nesting.

Reply:
Fixed